schema/constraint_object.go

```diff
@@ -25,6 +25,10 @@ type Object struct {
 
 	// Description defines description of the whole object (affects hover)
 	Description lang.MarkupContent
+
+	// AllowInterpolatedAttrName determines whether the attribute names can be
+	// interpolated (true) or static (literal strings only).
+	AllowInterpolatedAttrName bool
```

Comment:
While `AllowInterpolatedAttrName` is technically more correct, I would suggest using `AllowInterpolatedKeys` as the field name here. This makes it easier to read and understand, and we have a single field/concept to enable the same behavior for maps and objects. In other places, like the comment in line 19 or `rawObjectKeys`, we also use the term `keys` for objects.

Reply:
Strictly speaking, the key becomes attribute when it is treated as an object. Perhaps confusingly (yet intentionally), the reason `rawObjectKeys` uses the term _key_ is because the context is `hclsyntax.ObjectConsExpr`. To put it differently, HCL's term for `{}` is "object constant" whether it ends up being a map or an object. The possibly confusing part is that HCL uses terms as "key" and "value" (in spite of referring to it as object), so this was all mostly in alignment with upstream naming.

All that said, I am not opposed to renaming this to `AllowInterpolatedKeys`, while not strictly "correct" it's not also entirely incorrect and it's pragmatic.

I can commit that patch.